pkg/io/option.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io
+
+import (
+	"cloud.google.com/go/storage"
+	"gocloud.dev/blob"
+	"google.golang.org/api/googleapi"
+)
+
+// A WriterOption is an option for the opener Writer method
+type WriterOption interface {
+	Apply(*storage.Writer, *blob.WriterOptions)
```

Comment:
This must mention that implementations have to nilcheck both args

Reply:
Fixed, but commented on the new Apply method